plutus-playground-client/src/Chain/View.purs

```diff
@@ -189,35 +196,121 @@ balancesView sequenceId walletKeys m =
             , text ")"
             ]
         ]
-    , row_ (balanceView <$> Map.toUnfoldable m)
+    , table
+        [ classes
+            [ ClassName "table"
+            , tableBordered
+            , tableSmall
+            , ClassName "balances-table"
+            ]
+        ]
+        [ thead_
+            [ tr_
+                ( th [ rowSpan 2 ] [ text "Beneficial Owner" ]
+                    : foldMapWithIndex (\currency s -> [ th [ colSpan (Set.size s) ] [ text $ showCurrency currency ] ]) headings
+                )
+            , tr_
+                ( foldMap (foldMap tokenHeadingView) headings
+                )
+            ]
+        , tbody_
+            ( foldMap
+                ( \owner ->
+                    [ tr [ class_ $ beneficialOwnerClass owner ]
+                        ( th_
+                            [ beneficialOwnerView walletKeys owner ]
+                            : foldMapWithIndex
+                                ( \currency ->
+                                    foldMap
+                                      ( \token ->
+                                          let
+                                            _thisBalance :: forall m. Monoid m => Fold' m (Map BeneficialOwner Value) Int
```

Comment:
why does `m` need to be a `Monoid`?

Reply:
Good question. Well, it's the use of `preview` that makes it so. But I'll change it to the more general and readable alias: ` _thisBalance :: Traversal' (Map BeneficialOwner Value) Int`.

Thanks.